lib/fixed-points.nix

```diff
@@ -311,28 +311,68 @@ rec {
     );
 
   /**
-    Compose two extending functions of the type expected by 'extends'
-    into one where changes made in the first are available in the
-    'super' of the second
+    Composes two [`overlays`](#chap-overlays) and returns a single overlay function that combines them.
 
+    :::{.note}
+    The result is produced by using the update operator `//`.
+    This means nested values of previous overlays are not merged recursively.
+    :::
 
     # Inputs
 
     `f`
 
-    : 1\. Function argument
+    : The first `overlay` to apply. Its first argument is `final` and its second argument is `prev`.
 
     `g`
 
-    : 2\. Function argument
+    : The second `overlay` to apply. Its first argument is `final` and its second argument is `prev` with the result of `f` `//`-merged into it.
 
     `final`
 
-    : 3\. Function argument
+    : The fixpoint when the overlays are evaluated with [`fix`](#function-library-lib.fixedPoints.fix).
 
     `prev`
 
-    : 4\. Function argument
+    : The return value of the original function that is extended by the overlays `f` and `g`.
+      If `composeExtensions f g` is itself used in a composition of overlays, then any overlays that precede it can be considered to be part of "the original function".
+
+
+    # Type
+
+    ```
+    # Pseudo code
+    let
+      OverlayFn :: ( { ... } -> { ... } -> { ... } );
+    in
+      composeExtensions :: OverlayFn -> OverlayFn -> OverlayFn
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.fixedPoints.composeExtensions` usage example
+
+    ```nix
+    let
+      # Each overlay function has 'final' and 'prev' as arguments.
+      overlayA = final: prev: { b = final.c; c = 3; };
+      overlayB = final: prev: { c = 10; cPrev = prev.c; };
+
+      extensions = composeExtensions overlayA overlayB;
+
+      # Calculate the fixed point of all composed overlays.
+      # With an empty attribute set as the initial value.
+      fixedpoint = lib.fix (lib.flip extensions { } );
+
+    in fixedpoint
+    =>
+    {
+      b = 10;
+      c = 10;
+      cPrev = 3;
+    }
+    ```
+    :::
   */
   composeExtensions =
```

Comment:
@roberth Another idea. composeExtensions is used to implement composeManyExtensions, which takes a list instead of exactly two overlays.
Couldn't we just document composeManyExtensions [], and move composeExtensions into a `let` of `composeManyExtensions` 
All uses cases should be covered with composeManyExtensions ?